### DISCUSSION TITLE
Mod for --override argument

### DISCUSSION THREAD

**Discussion by JonNesovic** [Romanitho/Winget-AutoUpdate#244] [Ideas]
Would it be possible to add a mod to manually set the --override argument. for example:

File Name: Mod\Adobe.Acrobat.Reader.64-bit-override.txt

Contents: "/sAll /rs /msi EULA_ACCEPT=YES DISABLEDESKTOPSHORTCUT=1"

Thank you for Winget-AutoUpdate! it is a great tool!

**Comment by KnifMelti** [comment]
And, I thought I was done 🤔...
...but, great thinking!
@Romanitho has it in [Winget-Install](https://github.com/Romanitho/Winget-Install#use-winget-native-parameters) but in WAU it has to be done in a mod, as you proposed @Nesovj, or...

I'll take a look at it!

**Reply by Romanitho** [reply]
Hi, i agree, it could be interesting. But when you install an app once, normally updates should not change anything in the config right ?

**Reply by JonNesovic** [reply]
I installed Adobe Reader (ID Adobe.Acrobat.Reader.64) with the arguments "EULA_ACCEPT=YES DISABLEDESKTOPSHORTCUT=1" but after upgrading, users get the EULA pop up and a desktop shortcut is created. Just to make sure I will replicate it and get back to you.

**Reply by KnifMelti** [reply]
The latest Adobe.Acrobat.Reader.64 in the winget repo is from Oct 11 (22.003.20258) and that version had a bug fixed in the version from Oct 22 (22.003.20263) leaving the shortcut on the desktop (https://www.adobe.com/devnet-docs/acrobatetk/tools/ReleaseNotesDC/continuous/dccontinuousoct2022qfe.html#bug-fixes)...
...I now have a working POC (using CDBurnerXP as an example) with a **Canneverbe.CDBurnerXP-override.txt** in mods:
![image](https://user-images.githubusercontent.com/102996177/206502023-089c83a7-2cc3-4a05-b99f-8ece9355ca84.png)

**Reply by KnifMelti** [reply]
#245

**Reply by KnifMelti** [reply]
...and, in the case of **Canneverbe.CDBurnerXP** anyway; it doesn't matter if it was installed with `ADDLOCAL=All REMOVE=Desktop_Shortcut /qn` the first time, when updating the desktop shortcut still gets installed (if not modded).

**Reply by Romanitho** [reply]
👏

**Comment by JonNesovic** [comment]
Hmm Maybe my issue is just Adobe Reader. I install one version back (

winget install --id Adobe.Acrobat.Reader.64 --version 22.002.20212 --override "switches"
then
winget upgrade --id Adobe.Acrobat.Reader.64 and get:

A newer version was found, but the install technology is different from the current version installed. Please uninstall the package and install the newer version.

I tried without --override and adobe 32 bit as well. Same error. I will try another package to see if --override makes a difference.

**Reply by KnifMelti** [reply]
When it comes to **WAU** and upgrading it first tries `winget upgrade` and if that fails it tries `winget install`

**Comment by JonNesovic** [comment]
Oops wasn't running in system context. Can confirm that the --override does not transfer from install to upgrade.

No desktop shortcut or EULA:
`& $winget install --exact --id Adobe.Acrobat.Reader.64-bit --version 22.002.20212 --accept-package-agreements --accept-source-agreements --scope=machine --override "/sAll /rs /msi EULA_ACCEPT=YES DISABLEDESKTOPSHORTCUT=1"`

After upgrade desktop shortcut and EULA are present
`& $winget upgrade --exact --id Adobe.Acrobat.Reader.64-bit --accept-package-agreements --accept-source-agreements`

But if I run the upgrade with --override, no EULA or shortcut
`& $winget upgrade --exact --id Adobe.Acrobat.Reader.64-bit --accept-package-agreements --accept-source-agreements --override "/sAll /rs /msi EULA_ACCEPT=YES DISABLEDESKTOPSHORTCUT=1"`

So, I think the override mod will come in handy for suppressing reboots, shortcuts, prompts, etc.
